api/v1alpha1/initialbuildpipelineselector_types.go

```diff
@@ -0,0 +1,145 @@
+
+	// Defines if a Dockerfile should be present in the component, e.g. 'true'.
+	// The value to compare with is taken from devfile components of image type.
+	// +kubebuilder:validation:Optional
+	DockerfileRequired BoolString `json:"dockerfile,omitempty"`
+
+	// Defines names of component to match, e.g. 'my-component'.
+	// The value to compare with is taken from component.metadata.name field.
+	// +kubebuilder:validation:Optional
+	ComponentName string `json:"componentName,omitempty"`
+
+	// Defines annotations to match.
+	// The values to compare with are taken from component.metadata.annotations field.
+	// +kubebuilder:validation:Optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Defines labels to match.
+	// The values to compare with are taken from component.metadata.labels field.
+	// +kubebuilder:validation:Optional
+	Labels map[string]string `json:"labels,omitempty"`
+}
+
+// PipelineSelector defines allowed build pipeline and conditions when it should be used.
+type PipelineSelector struct {
+	// Name of the selector item. Optional.
+	// +kubebuilder:validation:Optional
+	Name string `json:"name,omitempty"`
+
+	// Build Pipeline to use if the selector conditions are met.
+	// +kubebuilder:validation:Required
+	PipelineRef tektonapi.PipelineRef `json:"pipelineRef"`
+
+	// Extra arguments to add to the specified pipeline run.
+	// +kubebuilder:validation:Optional
+	// +listType=atomic
+	PipelineParams []tektonapi.Param `json:"pipelineParams,omitempty"`
+
+	// Defines the selector conditions when given build pipeline should be used.
+	// All conditions are connected via AND, whereas cases within any condition connected via OR.
+	// If the section is omitted, then the condition is considered true (usually used for fallback condition).
+	// +kubebuilder:validation:Optional
+	WhenConditions WhenCondition `json:"when,omitempty"`
+}
+
+// InitialBuildPipelineSelectorSpec defines the desired state of InitialBuildPipelineSelector
+type InitialBuildPipelineSelectorSpec struct {
+	// Defines chain of pipeline selectors.
+	// The first matching item is used.
+	// +kubebuilder:validation:Required
+	Selectors []PipelineSelector `json:"selectors"`
+}
+
+// InitialBuildPipelineSelectorStatus defines the observed state of InitialBuildPipelineSelector
+type InitialBuildPipelineSelectorStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	// TODO
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// InitialBuildPipelineSelector is the Schema for the initialbuildpipelineselectors API
+type InitialBuildPipelineSelector struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   InitialBuildPipelineSelectorSpec   `json:"spec,omitempty"`
+	Status InitialBuildPipelineSelectorStatus `json:"status,omitempty"`
```

Comment:
not sure if we will need status, it's basically static object.

Reply:
I am not sure either. We might need it for reporting some issues with the config... until then we might have it empty or delete completely.